src/System.Security.Cryptography.Algorithms/tests/System.Security.Cryptography.Algorithms.Tests.csproj

```diff
@@ -3,6 +3,7 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <ProjectGuid>{508A7D81-6462-459C-9F8F-B58FCCCFC8E7}</ProjectGuid>
+    <DefineConstants>$(DefineConstants);TESTING_CSP_IMPLEMENTATION</DefineConstants>
```

Comment:
This library is CSP based on netfx.  CNG based on Windows with corefx, OpenSSL-based with Linux, and Security.framework-based on macOS.

So the \#define should be as scoped as it can get.